src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js

```diff
@@ -21,7 +21,7 @@ const startTimer = () => {
 
 function BaseSidebarScreen(props) {
     const styles = useThemeStyles();
-    const {activeWorkspaceID} = useActiveWorkspace();
+    const activeWorkspaceID = Navigation.getPolicyIDFromNavigationState();
```

Comment:
Do we still need the `useActiveWorkspace` context in the ideal nav v2?

Reply:
I think we need, because getting this value from the context should be faster than invoking this new method, because it has complex logic. In this place in code we need to get this value after update immediately but in other cases context should work fine. cc: @adamgrzybowski

Reply:
After some thought, I think we should check the following:
- Is this method really slower? This is something that should be easy to check with `console.time`. Make sure to do this when the navigation state is big. 
- If we use this hook somewhere to trigger rerender. 

If the answer is no and no then I would consider removing the context. It made sense to have it before ideal-nav-v2. Not sure about the current situation.

Keeping only one method to access policyID of the switcher may prevent confusion in the future.

Reply:
Yes, I've checked it in more detail, and the context works faster than this method. Here is necessary because we need to use a new policyID immediately after update, but in other cases it's better to use the context. 

```
   const handleStateChange = (state: NavigationState | undefined) => {
        if (!state) {
            return;
        }
        const activeWorkspaceID = getPolicyIDFromState(state as NavigationState<RootStackParamList>);
        // Performance optimization to avoid context consumers to delay first render
        setTimeout(() => {
            currentReportIDValue?.updateCurrentReportID(state);
            setActiveWorkspaceID(activeWorkspaceID);
        }, 0);
        parseAndLogRoute(state);
    };
```

This is a method in the `NavigationRoot.tsx` file that updates the value stored in the context. In this particular case we need to read it faster and we can't wait for the update of this value, so we read it from the navigation state, but in other cases where we don't need to display this value it's fine.